### ISSUE TITLE
Is there a way to interrupt/cancel a long running query?

### ISSUE THREAD

**Issue by johann-petrak** [pymssql/pymssql#629]
Various GUIs for the MSSQL server allow to abort a running query. What is the correct way to do this with pymssql?

**Comment by rsyring** [comment]
The issue tracker isn't a great place to ask support questions.  Please try StackOverflow or the mailing list: https://groups.google.com/forum/#!forum/pymssql

Thanks.

**Comment by johann-petrak** [comment]
I guess I tried to word this too politely: there seems to be no support or documentation for this so it should probably have been a feature request for either implementing it or documenting it if it is already implemented.

**Comment by rsyring** [comment]
Fair enough, I've re-opened the issued.  Please update the subject/body to note this is a feature request.  However, just so you are aware, pymssql struggles to get active maintenance, so feature requests are likely a long way off: https://github.com/pymssql/pymssql/issues/516

**Comment by johann-petrak** [comment]
Various GUIs (Microsoft's own, dbeaver, ..) allow to interrupt/cancel a long running query. Apparently there is a protocol for the client notifying the server that the query got interrupt which the server honours as soon as possible (in most cases where interrupting the query does not require a transaction to get rolled back, immediately).
It would be extremely useful if this is possible through pymssql as well.

**Comment by cam72cam** [comment]
I've found that calling conn.cancel() works quite well from the same thread most of the time but have seen some segfaults occasionally (using gevent)

If called from a different thread it's pretty much guaranteed to segfault.

**Comment by sergioalvarezlavin** [comment]
> Various GUIs for the MSSQL server allow to abort a running query. What is the correct way to do this with pymssql?

@johann-petrak I understand you want to cancel a query which is being executed. I found a solutution using pymssql but it may affect to all your code related to the database.
`pymssql` has two modules/subclasses `_pymssql` and `_mssql`. I realized that, normally, the class uses the `_pymssql` and this module has no cancel function. However, if you force to use the `_mssql` you have this option. 
As I mentioned before, this may require to update your code because the functions some functions will be different, e.g., connection (different arguments), cursor (will not exist). You can find the solution in the documentation from pymssl. 

I hope this helps.
